src/frontend/src/repositories/anchorMetadata.ts

```diff
@@ -0,0 +1,194 @@
+import { MetadataMapV2 } from "$generated/internet_identity_types";
+
+export type AnchorMetadata = {
+  recoveryPageShownTimestampMillis?: number;
+};
+
+export const RECOVERY_PAGE_SHOW_TIMESTAMP_MILLIS =
+  "recoveryPageShownTimestampMillis";
+
+const convertMetadata = (rawMetadata: MetadataMapV2): AnchorMetadata => {
+  const recoveryPageEntry = rawMetadata.find(
+    ([key]) => key === RECOVERY_PAGE_SHOW_TIMESTAMP_MILLIS
+  );
+  if (recoveryPageEntry === undefined) {
+    return {};
+  }
+  const stringValue = recoveryPageEntry[1];
+  if ("String" in stringValue) {
+    const recoveryPageShownTimestampMillis = Number(stringValue.String);
+    if (isNaN(recoveryPageShownTimestampMillis)) {
+      return {};
+    }
+    return {
+      recoveryPageShownTimestampMillis,
+    };
+  }
+  return {};
+};
+
+type MetadataGetter = () => Promise<MetadataMapV2>;
+type MetadataSetter = (metadata: MetadataMapV2) => Promise<void>;
+type RawMetadataState = MetadataMapV2 | "loading" | "error" | "not-loaded";
+
+/**
+ * Class to manage the metadata of the anchor and interact with the canister.
+ *
+ * We decided to not throw any errors because this is non-critical for the application
+ * and we don't want to disrupt user flows if there is an error with the metadata.
+ *
+ * This class loads the metadata in the background when it's created.
+ * It can then be read and updated in memory.
+ * The metadata needs to be committed to the canister with the `commitMetadata` method to persist the changes.
+ */
+export class AnchorMetadataRepository {
```

Comment:
Quite a while ago we decided to sunset the term "anchor" because it is confusing. So on the UI, this term is no longer used. In the code, we didn't change things.
But for new code, we now use "identity" rather than anchor. Hence, why on the API level it is called `IdentityMetadata` rather than `AnchorMetadata`.
I think it would be better to use the new terminology here too and keep going with the slow journey of eventually replacing all the occurrences of "anchor".

Reply:
Done!